### PR TITLE
Generalize CI configs

### PR DESCRIPTION
**Pull request by tamird**
Avoid encoding x64,386,armv{6,7},aarch64 in so many places.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ on:
   pull_request:
     branches:
       - master
+  schedule:
+    - cron: 00 4 * * *
```
Comment (petermattis) [inline .github/workflows/go.yml:11]: Can you elaborate on why this is beneficial? What sort of breakage are you imaging will be detected before a PR is sent? No real objection, just curious to understand what I'm missing.

**Review by petermattis — COMMENTED** [review]
LGTM, though I really have no idea what most of these changes are doing.

File: .github/workflows/go.yml (hunk 1 above)
Comment (tamird) [reply .github/workflows/go.yml:11]: One example is highlighted in https://github.com/petermattis/goid/pull/45/commits/2b7fc54c033862f907740533695a6f7649be23f7 - gccgo-13 was removed at some point and we didn't know until #44.

Comment (petermattis) [reply .github/workflows/go.yml:11]: Ah, got it. Thanks.

**Pull request merged**